src/rest-server/src/templates/yarnContainerScript.mustache

```diff
@@ -30,9 +30,33 @@ BASH_XTRACEFD=13
 function exit_handler()
 {
   rc=$?
-  printf "[DEBUG] EXIT signal received in yarn container, performing clean up action...\n"
+  echo "Exited with $rc"
+  local handler="Yarn container exit handler"
+  debug_log "$handler"  "EXIT signal received in yarn container, performing clean up action..."
+
+  debug_log "$handler" "trying to kill docker container $docker_name"
+  docker logs $docker_name
+  docker inspect $docker_name
+  pid=$(docker inspect --format={{{ inspectPidFormat }}} $docker_name 2>/dev/null)
+  if [ $pid -gt 0 ]; then
+    kill -9 $pid &&\
+      debug_log "$handler" "docker caontainer $docker_name killed successfully." ||\
+      debug_log "$handler" "tries to kill the container $docker_name but failed. Maybe it has already exited."
+  else
+    debug_log "$handler" "docker container $docker_name has already exited"
+    is_oom=$(docker inspect --format={{{ inspectOOMKilledFormat }}} $docker_name 2>/dev/null)
+    debug_log "$handler" "docker container $docker_name is exited by OOM? $is_oom"
+    if [ "$is_oom" = "true" ]; then
+      rc=55
+    fi
+  fi
 
-  printf "[DEBUG] Write exit code $rc to file /var/lib/hadoopdata/nm-local-dir/nmPrivate/$APP_ID/$CONTAINER_ID/$CONTAINER_ID.pid.exitcode.\n"
+  docker container rm $docker_name
```

Comment:
@Gerhut 
We can't guarantee exit_handle will be executed, all codes here are besteffort. So the container might be left on the host.

Reply:
You're right, but there is an abnormal state when exit_handle does not be executed. The legacy containers removing should be rely on cleaner.